biz.aQute.launcher/src/aQute/launcher/Launcher.java

```diff
@@ -650,22 +671,80 @@ private String toNativePath(String s) {
 	 * @throws IOException
 	 */
 	void installEmbedded(List<Bundle> tobestarted) throws BundleException, IOException {
+		Manifest m = EmbeddedLauncher.MANIFEST;
 		trace("starting in embedded mode");
 		BundleContext context = systemBundle.getBundleContext();
 		for (Object o : parms.runbundles) {
 			String path = (String) o;
-			trace("installing %s", path);
+			String sha = m.getAttributes(path).getValue("SHA-Digest");
+
 			try (InputStream in = getClass().getClassLoader().getResourceAsStream(path)) {
 				Bundle bundle = getBundleByLocation(path);
-				if (bundle == null)
+				if (bundle == null) {
+					trace("installing %s", path);
 					bundle = context.installBundle(path, in);
-				else
-					bundle.update(in);
+				} else {
+					if (mustUpdate(sha, bundle)) {
+						trace("updating %s, sha=%s", path, sha);
+						bundle.stop();
+						bundle.update(in);
+					} else {
+						trace("not updating %s because identical sha=%s", path, sha);
+					}
+				}
+				updateSha(sha, bundle);
 				tobestarted.add(bundle);
 			}
 		}
 	}
 
+	/*
+	 * Check if we have a SHA from the manifest and it it was for this the
+	 * bundle.
+	 */
+	private boolean mustUpdate(String sha, Bundle bundle) {
+		if (sha == null)
+			return true;
+
+		File shafile = shafile(bundle);
+
+		if (shafile == null || !shafile.isFile() || !shafile.canRead())
+			return true;
+
+		try {
+			String storedSha = IO.collect(shafile);
+			if (storedSha == null || !storedSha.equals(sha))
```

Comment:
Shouldn't this use `equalsIgnoreCase` to be safe? Since you are using the manifest header value and its case is not defined.